cmake/sdlchecks.cmake

```diff
@@ -583,6 +583,26 @@ macro(CheckWayland)
   endif()
 endmacro()
 
+macro(CheckLibDonnell)
+  if(SDL_LIBDONNELL)
+    set(HAVE_LIBDONNELL FALSE)
+	set(LibDonnell_PKG_CONFIG_SPEC donnell)
+	pkg_check_modules(PC_LIBDONNELL IMPORTED_TARGET ${LibDonnell_PKG_CONFIG_SPEC})
+
+    if(PC_LIBDONNELL_FOUND)
+        set(HAVE_LIBDONNELL TRUE)
+		set(HAVE_LIBDONNELL_H TRUE)
+        cmake_push_check_state()
+        list(APPEND CMAKE_REQUIRED_INCLUDES ${PC_LIBDONNELL_INCLUDE_DIRS})
+        list(APPEND CMAKE_REQUIRED_LIBRARIES PkgConfig::PC_LIBDONNELL)
+        cmake_pop_check_state()
+        FindLibraryAndSONAME("donnell" LIBDIRS ${PC_LIBDONNELL_LIBRARY_DIRS})
```

Comment:
This is also unneeded, since you are not loading `libdonnell.so` dynamically.
Because we link to `libdonnell.so`, running `ldd libSDL3.so` returns a huge list:
```
[maarten@fedora cmake-build-debug]$ ldd libSDL3.so
	linux-vdso.so.1 (0x00007ffd8735c000)
	libm.so.6 => /lib64/libm.so.6 (0x00007fd90692d000)
	libdonnell.so => /tmp/libdonnell-release1.1/prefix/lib/libdonnell.so (0x00007fd906499000)
	libc.so.6 => /lib64/libc.so.6 (0x00007fd9062bb000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fd906a32000)
	libfribidi.so.0 => /lib64/libfribidi.so.0 (0x00007fd90690e000)
	libfontconfig.so.1 => /lib64/libfontconfig.so.1 (0x00007fd90626c000)
	libfreetype.so.6 => /lib64/libfreetype.so.6 (0x00007fd90619a000)
	libpng16.so.16 => /lib64/libpng16.so.16 (0x00007fd906161000)
	libz.so.1 => /lib64/libz.so.1 (0x00007fd906147000)
	libunistring.so.5 => /lib64/libunistring.so.5 (0x00007fd905f98000)
	libxml2.so.2 => /lib64/libxml2.so.2 (0x00007fd905e24000)
	libbz2.so.1 => /lib64/libbz2.so.1 (0x00007fd905e10000)
	libharfbuzz.so.0 => /lib64/libharfbuzz.so.0 (0x00007fd905d12000)
	libbrotlidec.so.1 => /lib64/libbrotlidec.so.1 (0x00007fd905d05000)
	liblzma.so.5 => /lib64/liblzma.so.5 (0x00007fd905cd2000)
	libglib-2.0.so.0 => /lib64/libglib-2.0.so.0 (0x00007fd905b89000)
	libgraphite2.so.3 => /lib64/libgraphite2.so.3 (0x00007fd905b68000)
	libbrotlicommon.so.1 => /lib64/libbrotlicommon.so.1 (0x00007fd905b43000)
	libpcre2-8.so.0 => /lib64/libpcre2-8.so.0 (0x00007fd905aa9000)
```
instead of this before:
```
[maarten@fedora cmake-build-debug]$ ldd libSDL3.so
	linux-vdso.so.1 (0x00007fff004ae000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f0c1b7a4000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f0c1b222000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f0c1b8a9000)
```